### PR TITLE
wip

### PR DESCRIPTION
**Pull request by andymckay**
* this is a wip for #668 
* tries to differentiate the errors coming back to seperate out required errors vs array errors and so will hopefully cope with some of the more complicated ones we haven't implemented yet, like content scripts

screenshot 2016-05-12 19 51 06

* feedback gratefully received and yeah no tests yet

### CODE REVIEW

**Comment by andymckay** [comment]
... and if its not an array, we get a fatal error

screenshot 2016-05-12 19 53 33

File: src/parsers/manifestjson.js [hunk 1]
```diff
@@ -37,28 +37,48 @@ export default class ManifestJSONParser {
     this.isValid = this._validate();
   }
 
+  errorLookup(error) {
+    var result = {
```
Comment (tofumatt) [inline src/parsers/manifestjson.js]: This is odd as it appears to be a message that isn't in the `message.js` file. We kept everything there so we'd be able to trace where all messages came from. I'd start with a message there and add your dynamically built `message` attribute to it here.

**Comment by tofumatt** [comment]
I had a quibble with the location of the message data but I'm 👍🏻 on the idea! I like the output a *lot* more with your implementation.

**Comment by andymckay** [comment]
I ended with a new pull request, oops.

**Pull request closed without merging**